### PR TITLE
add nrel integration

### PR DESCRIPTION
**Pull request by ros3bud**
This is a new outbound integration. The specs and overview can be found here: https://docs.google.com/document/d/1J5ODJmFUDh9p1rvBRClzjiAy0sfmOYveW25CNBhQltE/edit#heading=h.l1qkin6rkzz6.

LGTY?

### CODE REVIEW

**Comment by cgrayson** [comment]
Here's my first thoughts, all in a dump instead of comments per line. Let me know if you have questions on anything. 

- package.json missing `description` 
- Travis build not configured yet
- add a link to NREL in the Readme.md 
- `outbound.coffee`
	- in `validate()`, that message is difficult to understand 
		- also, I assume that’s the correct minimum (though I don’t see it anywhere in the overview or API docs) 
		- the presence of `api_key` should be checked as well
	- in `request()`, the local variable `req` is unused 
		- minor, but also: it’s a query string, not a body 
		- did you consider using the `X-Api-Key` header format for auth? might make looking at requests a little cleaner to not have the key included in every request
	- should the request variable `nrel.api_key` be `credential` type? 
	- in `response()`, the local variable `outputs` may as well be defined up one line
		- is there a chance that `body.outputs` is undefined? 
		- the final else sets `outcome` to “failure” but sets no `reason`
	- in the response variables, I think we usually omit ending periods in descriptions
		- I wonder if the last three would be more clear with `_rate` on the end of their names
		- and I think they should have the “($/kWh)” in their descriptions
		- also, should those be `number` types? 
		- I don’t see `census_region` in the API docs…? 
		- was it purposely decided to only capture `utility_name`s, and not `utility_id`s?
		- did you consider making `utility_name` an array? 
		- should it be named `utility_names` (plural)? 
	- since `getAddress()` is a one-liner, only called from one place, and not exported, I don’t think it needs to be a separate function 
- `outbound_spec.coffee`
	- fix indentation of `describe`s
	- declaration of `@vars` could be clearer and more standard coffeescript style
	- also, `vars` is local; doesn’t need to be `@`
	- since you’re testing the entire URL, separate tests for `city`, `state`, etc. aren’t needed 
	- unused variables `vars` and `req` on lines 51 & 52 
		- and 117 & 118
		- and 150 & 151 
	- in the “failure” case, probably don’t need `beforeEach` when there’s only a single test clause

**Comment by ros3bud** [comment]
Feedback incorporated. Also, we're going to use the /utility_rates/ endpoint, rather than /census_rate/, so I've removed the census_region response variable and test.

File: src/outbound.coffee [hunk 1]
```diff
@@ -1,45 +1,30 @@
-_s = require('underscore.string')
+#
+# Validate ---------------------------------------------------------------
+#
 
 
+validate = (vars) ->
+  lead = vars.lead
+  return 'either postal_code or both city and state must be present' unless lead.postal_code or (lead.city and lead.state)
+  return 'nrel.api_key must not be blank' unless vars.nrel?.api_key
```
Comment (cgrayson) [inline src/outbound.coffee]: These would be a little clearer if they were given in the prettified version the user sees in the UI (e.g., "NREL API key").

File: src/outbound.coffee [hunk 2]
```diff
@@ -78,23 +56,28 @@ request.variables = ->
...
+    result.sources = body.metadata?.sources
+  else if res.status isnt 200
+      result.outcome = 'error'
+      result.reason =  nrelError or "NREL error (#{res.status})"
   else
-    reason: errorReason
-    outcome: 'error'    
- 
+    result.outcome = 'failure'
+    result.reason = 'utility company name(s) unavailable'
```
Comment (cgrayson) [inline src/outbound.coffee]: I wonder if "unavailable" might be mistaken for the service itself being unavailable, and if something like "No utility information found" would be clearer.

File: src/outbound.coffee [hunk 3]
```diff
@@ -102,79 +85,18 @@ response = (vars, req, res) ->
 #
 response.variables = ->
   [
-    { name: 'outcome', type: 'string', description: 'Integration outcome (success, failure, or error)' },
-    { name: 'reason', type: 'string', description: 'If outcome is error or failue, the error reason' },
-    { name: 'redirect_url',  type: 'url', description: 'The redirect URL provided by Lendio' }
+    { name: 'outbound.outcome', type: 'string', description: 'Integration outcome (success, failure, or error)' },
+    { name: 'outbound.reason', type: 'array', description: 'If outcome is error or failue, the error reason(s)' },
+    { name: 'outbound.utility_names', type: 'array', description: 'The name(s) of the utility company' },
+    { name: 'outbound.company_ids', type: 'array', description: 'The ID(s) of the utility company, returned as a comma-delimited string' },
```
Comment (cgrayson) [inline src/outbound.coffee]: This isn't a comma-delimited string anymore.

**Comment by cgrayson** [comment]
Otherwise LGTM :+1:

File: src/outbound.coffee (hunk 1 above)
Comment (ros3bud) [inline src/outbound.coffee]: Good idea.

File: src/outbound.coffee (hunk 2 above)
Comment (ros3bud) [inline src/outbound.coffee]: Yes, I think so.

File: src/outbound.coffee (hunk 3 above)
Comment (ros3bud) [inline src/outbound.coffee]: Got it.

**Pull request merged**